textX-LS/core/textx_ls_core/utils.py

```diff
@@ -85,8 +85,25 @@ def get_project_name_and_version(folder_or_wheel: str) -> Tuple[str, str]:
         with mock.patch.object(setuptools, "setup") as mock_setup:
             with open(setuppy, "r") as f:
                 exec(f.read())
-            _, kwargs = mock_setup.call_args
-        project_name, version = kwargs.get("name", None), kwargs.get("version", None)
+            _, metadata = mock_setup.call_args
+
+        if "name" not in metadata:
```

Comment:
@danixeee Sometimes `name` is given in `setup.cfg` and `version` in `setup()` (e.g. `version` is loaded from module).
I suggest parsing `setup.cfg` if exists and doing something like:
```python
metadata_cfg = dict(cfg_parser.items("metadata"))
project_name, version = (
    metadata.get("name", metadata_cfg.get("name", None)),
    metadata.get("version", metadata_cfg.get("version", None)))
```

Reply:
@igordejanovic  I didn't know that, thanks. I will create an empty _metadata_ dict and update with values from both `setup.py` and `setup.cfg` and read `name` and `version` after.